controller/uninstall_controller.go

```diff
@@ -770,16 +770,17 @@ func (c *UninstallController) deleteReplicas(replicas map[string]*longhorn.Repli
 
 // deleteLeftBackups deletes the backup having no backup volume
 func (c *UninstallController) deleteLeftBackups(backup *longhorn.Backup) (err error) {
-	volumeName, ok := backup.Labels[types.LonghornLabelBackupVolume]
-	if !ok {
+	volumeName, volumeNameExists := backup.Labels[types.LonghornLabelBackupVolume]
+	backupTargetName, backupTargetNameExists := backup.Labels[types.LonghornLabelBackupTarget]
+	if !volumeNameExists || !backupTargetNameExists {
 		// directly delete it if there is even no backup volume label
 		if err = c.ds.DeleteBackup(backup.Name); err != nil {
 			if !apierrors.IsNotFound(err) {
 				return errors.Wrapf(err, "failed to delete backup %v", backup.Name)
 			}
 		}
 	}
-	_, err = c.ds.GetBackupVolumeByBackupTargetAndVolumeRO(backup.Status.BackupTargetName, volumeName)
+	_, err = c.ds.GetBackupVolumeByBackupTargetAndVolumeRO(backupTargetName, volumeName)
```

Comment:
Any reason we can't use `backup.Status.BackupTargetName`? Is it possible that the label of backkup target is a different from the status?

Reply:
The failed backup target will not update the `backup.Status.xxxx` so we can not use this field `backup.Status.BackupTargetName` to get the backup target name.
I will revert it and do some tests because I have modified the method `GetBackupVolumeByBackupTargetAndVolumeRO` to return the `NotFound` error if the backup target name is an empty string.

Reply:
Revert. Thanks